src/components/PopupWithForm.js

```diff
@@ -1,13 +1,19 @@
 import Popup from "../components/Popup.js";
+
 export class PopupWithForm extends Popup {
-  constructor(popup, { handleFormSubmit }) {
+  constructor(popup, { handleFormSubmit, handleClearError, renderLoading }) {
     super(popup);
     this.handleFormSubmit = handleFormSubmit;
     this.form = this.popup.querySelector(".form");
+
+    this.handleClearError = handleClearError;
+    this._renderLoading = renderLoading;
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```